funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_manager.py

```diff
@@ -274,44 +275,10 @@ def pull_tasks(self, kill_event):
                 self.task_incoming.send(msg)
 
             # Receive results from the workers, if any
-            socks = dict(poller.poll(timeout=poll_timer))
+            socks = dict(self.poller.poll(timeout=poll_timer))
+
             if self.funcx_task_socket in socks and socks[self.funcx_task_socket] == zmq.POLLIN:
-                try:
-                    w_id, m_type, message = self.funcx_task_socket.recv_multipart()
-                    if m_type == b'REGISTER':
-                        reg_info = pickle.loads(message)
-                        logger.debug("Registration received from worker:{} {}".format(w_id, reg_info))
-                        self.worker_map.register_worker(w_id, reg_info['worker_type'])
-
-                    elif m_type == b'TASK_RET':
-                        logger.debug("Result received from worker: {}".format(w_id))
-                        logger.debug("[TASK_PULL_THREAD] Got result: {}".format(message))
-                        self.pending_result_queue.put(message)
-                        self.worker_map.put_worker(w_id)
-                        task_done_counter += 1
-                        task_id = pickle.loads(message)['task_id']
-                        task_type = self.task_type_mapping.pop(task_id)
-                        self.task_status_deltas.pop(task_id, None)
-                        logger.debug("Task type: {}".format(task_type))
-                        self.outstanding_task_count[task_type] -= 1
-                        logger.debug("Got result: Outstanding task counts: {}".format(self.outstanding_task_count))
-
-                    elif m_type == b'WRKR_DIE':
-                        logger.debug("[WORKER_REMOVE] Removing worker {} from worker_map...".format(w_id))
-                        logger.debug("Ready worker counts: {}".format(self.worker_map.ready_worker_type_counts))
-                        logger.debug("Total worker counts: {}".format(self.worker_map.total_worker_type_counts))
-                        self.worker_map.remove_worker(w_id)
-                        proc = self.worker_procs.pop(w_id.decode())
-                        if not proc.poll():
-                            try:
-                                proc.wait(timeout=1)
-                            except subprocess.TimeoutExpired:
-                                logger.warning(f"[WORKER_REMOVE] Timeout waiting for worker {w_id} process to terminate")
-                        logger.debug(f"[WORKER_REMOVE] Removing worker {w_id} process object")
-                        logger.debug(f"[WORKER_REMOVE] Worker processes: {self.worker_procs}")
-
-                except Exception as e:
-                    logger.exception("[TASK_PULL_THREAD] FUNCX : caught {}".format(e))
+                _ = self.poll_funcx_task_socket(task_done_counter)
```

Comment:
the `task_done_counter` is not updated correctly. Note above `pending_task_count = task_recv_counter - task_done_counter`, `task_done_counter` is used to decide task advertisement. So this value needs to be updated carefully and correctly.